internal/helmdeployer/deployer.go

```diff
@@ -246,30 +246,40 @@ func (h *Helm) Deploy(bundleID string, manifest *manifest.Manifest, options flee
 
 // RemoveExternalChanges does a helm rollback to remove changes made outside of fleet.
 // It removes the helm history entry if the rollback fails.
-func (h *Helm) RemoveExternalChanges(bd *fleet.BundleDeployment) error {
+func (h *Helm) RemoveExternalChanges(bd *fleet.BundleDeployment) (*Resources, error) {
 	logrus.Infof("Drift correction: rollback BundleDeployment %s", bd.Name)
 
 	_, defaultNamespace, releaseName := h.getOpts(bd.Name, bd.Spec.Options)
 	cfg, err := h.getCfg(defaultNamespace, bd.Spec.Options.ServiceAccount)
 	if err != nil {
-		return err
+		return nil, err
 	}
 	currentRelease, err := cfg.Releases.Last(releaseName)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
 	r := action.NewRollback(&cfg)
 	r.Version = currentRelease.Version
+	r.MaxHistory = bd.Spec.Options.Helm.MaxHistory
+	if r.MaxHistory == 0 {
+		r.MaxHistory = MaxHelmHistory
+	}
 	if bd.Spec.CorrectDrift.Force {
 		r.Force = true
 	}
-	err = r.Run(releaseName)
-	if err != nil && !bd.Spec.CorrectDrift.KeepFailHistory {
-		return removeFailedRollback(cfg, currentRelease, err)
+	if err = r.Run(releaseName); err != nil {
+		if bd.Spec.CorrectDrift.KeepFailHistory {
+			return nil, err
+		}
+		return nil, removeFailedRollback(cfg, currentRelease, err)
 	}
 
-	return err
+	release, err := cfg.Releases.Last(releaseName)
+	if err != nil {
+		return nil, err
+	}
+	return releaseToResources(release)
```

Comment:
This involves converting the full contents of the release manifest into YAML objects; I wonder if that might be a bit overkill, when all we need is the release ID. It could make sense to extract release name computation into a separate function instead, and call that.

Reply:
Good point, addressed in 235d80be